### PR TITLE
Add a back button to various views on the site

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context

The GOV.UK design system states that we should offer a back link in form pages. 

The back functionality is implemented via JS, as way to toggle back in Django as only the immediate referrer page is recorded. Another solution might be to store a user view history, but this would probably be too substantial a task. For users without JS, the back link is hidden, and the browser back button is a fallback option.

## Changes proposed in this pull request

Adds a GOV.UK Back link https://design-system.service.gov.uk/components/back-link/ to specific views. 

back link

### Views with a back link
* Form views, so users can change previous answers
* Evaluation detail view, so users can quickly return to the search page (with filters and page number maintained) if they are navigating between different Evaluations.

### Views without a back link
* Signed-out view, as we want to be directing users only to sign in.
* Home page, as clicking 'back' could lead users back out of the site, and there are other clear navigation buttons given instead.
* Evaluation list view, as this will become the new home page when the site goes live.
* Form confirmation, as if users want to make changes, they would need to edit the newly-created Evaluation rather than make a new one.

## Guidance to review
* Click on the back links, are you taken where you expect?
* Are there any views that currently do not have back links that you think need them?
* Double-check the Back link guidance - have I misunderstood anything https://design-system.service.gov.uk/components/back-link/ ?
* There's a bit more repetition than I'd like in the templates - unfortunately, I couldn't get `macro` to play nicely with `block` - is there a better way of doing this?

## Link to JIRA ticket

https://technologyprogramme.atlassian.net/browse/ER-59

## Things to check

~- [ ] I have added any new ENV vars in all deployed environments~

### CODE REVIEW

File: evaluation_registry/templates/evaluation_list.html [hunk 1]
```diff
@@ -206,7 +206,7 @@ <h2 class="govuk-accordion__section-heading">
                   <div>
                     {% if is_authenticated %}
                       <div class="title-tagged">
-                        <a href="{{url('evaluation-detail', evaluation.id)}}?{{request.GET.urlencode()}}" class="govuk-link">
```
Comment (rachaelcodes) [inline evaluation_registry/templates/evaluation_list.html:209]: Removing this and in line 225 as it was part of a very rough back button implementation when this repo was just an Evaluation search tool

**Comment by nmenezes0** [comment]
I'm really confused, I can't see the "Back" links (I checked I was on the right branch!) - am I missing something?
image

I also expected to see them in the views you get when you are creating an evaluation (when you click on "Share evaluation" from the homepage), but I couldn't see the back link there either.

And the code looks fine to me. I haven't worked on this project much, so I could be looking in the wrong places.

**Comment by nmenezes0** [comment]
> I'm really confused, I can't see the "Back" links (I checked I was on the right branch!) - am I missing something? image
> 
> I also expected to see them in the views you get when you are creating an evaluation (when you click on "Share evaluation" from the homepage), but I couldn't see the back link there either.
> 
> And the code looks fine to me. I haven't worked on this project much, so I could be looking in the wrong places.

Must be something with my browser. I'm now using Safari and it looks fine!

**Review by nmenezes0 — APPROVED** [review]
Looks good - agree with your choice of pages that should have back links, think it aligns with the gov.uk back link guidance, not sure of how to write macro better.

**Pull request merged**